app/scripts/lib/marketing-email-client.js

```diff
@@ -53,7 +63,11 @@ define(function (require, exports, module) {
     },
 
     optIn (accessToken, newsletterId) {
-      return this._request('post', '/subscribe', accessToken, {
+      const cleanedSourceUrl =
+        Url.cleanSearchString(this._window.location.href, ALLOWED_SOURCE_URL_QUERY_PARAMS);
+      const encodedSourceUrl = encodeURIComponent(cleanedSourceUrl);
+      const url = `/subscribe?source_url=${encodedSourceUrl}`;
```

Comment:
Per [1], I believe `source_url` should to be passed in the POST body rather than in the query params.

[1] http://basket.readthedocs.io/newsletter_api.html#news-subscribe